core/jazz_splunk-kinesis-log-streamer/components/utils.js

```diff
@@ -55,47 +122,189 @@ var getSubInfo = function (message, patternStr, index) {
   return result;
 }
 
-var getCommonData = function (payload) {
+var getCommonData = function (payload, config) {
+  // first get token for calling respective APIs
+  getToken(config)
+  .then((creds) => {
+    // get configDB data for getting roleArn specific to account
+    getConfigJson(config, creds)
+    .then((configData) => {
+      // get accountId and region through service Data
+      getsServiceMetaData(config, payload.logGroup, creds)
+      .then((serviceData) => {
+        // execute sts:assumeRole
+        assumeRole(configData, serviceData)
+        .then((tempCreds) => {
+          // configure cloudwatch and retrieve tags from logGroup
+          getLogsGroupsTags(payload.logGroup, tempCreds, serviceData)
+          .then((tagsResult) => {
+            return new Promise((resolve, reject) => {
+              let data = {};
+              data.metadata = {};
+              data.asset_type = "lambda"
+              data.request_id = getInfo(payload.logEvents, global_config.PATTERNS.lambda_request_id);
+              if (data.request_id) {
+                data.provider = "aws_lambda";
+                let domainAndservice, serviceInfo, dev_environment, serviceInfoArr, namespace;
+                data.asset_identifier = payload.logGroup.split(global_config.PATTERNS.asset_identifier_key)[1];
+                // if tags present, then it is for sls-app, otherwise for other services
+                if(tagsResult != 'error' && tagsResult != undefined && tagsResult.tags.environment){
+                  data.environment = tagsResult.tags.environment;
+                  data.namespace = tagsResult.tags.namespace;
+                  data.service = tagsResult.tags.service;
+                } else {
+                  data.environment = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment, 2);
+                  if (data.environment === "dev") {
+                    dev_environment = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment_dev, 2);
+                    serviceInfo = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_environment_dev, 1);
+                    data.environment = dev_environment;
+                  } else {
+                    serviceInfo = getSubInfo(payload.logGroup, global_config.PATTERNS.Lambda_domain_service, 1);
+                  }
+                  domainAndservice = serviceInfo;
+                  if (serviceInfo.indexOf(global_config.PATTERNS.sls_app_function) !== -1) { // if yes then sls-app function
```

Comment:
Do we still need this `sls_app_function` block?

Reply:
No. Removed.